enhancements/operator-sdk/downstream-operator-sdk.md

```diff
@@ -0,0 +1,385 @@
+
+The downstream repo will contain a special _overlay_ branch that contains all of
+the downstream-specific changes. The files to be contained in this overlay are
+as follows:
+
+* Dockerfiles to support CI
+* Dockerfiles to build downstream images
+* Vendor Go dependencies
+* Ansible collections to support Ansible operator
+
+We will pin the `downstream:master` to `upstream:master` and
+_upstream_release_branches_ to _downstream_release_branches_. This means we will
+manage the OpenShift release branches ourselves. So `release-x.y` will track a
+specific upstream branch that was mirrored. Only tagged commits will ever be
+brought into the release branches.
+
+#### Syncing
+
+Today, we sync `operator-framework/operator-sdk` into
+`openshift/ocp-release-operator-sdk`. We sync it manually with `git merge`. The
+`master` branch downstream currently tracks the latest OpenShift release. So
+when moving to a new major SDK release downstream, the `master` branch sync is
+frought with conflicts that can cause missing changes or just make for a very
+tedious merge.
+
+The idea behind the autosyncing strategy is that the downstream branches will
+match the upstream branches 1 for 1. So downstream `master` will track upstream
+`master`. Downstream `vx.y.z` will track upstream `vx.y.z`. So the syncs should
+occur with minimal conflicts. These downstream branches will NOT contain any
+special downstream fixes, they are strictly mirrors of the upstream.
+
+The autosyncing will occur on a frequent basis configured in the
+[configuration file][auto-config]. The frequency will be determined by our
+needs.
+
+We will manage our own OpenShift release branches. These branches will be a
+combination of a tagged upstream branch plus the overlay branch.
+
+There is an initial prototype of the syncing code already:
+
+https://github.com/fabianvf/ocp-osdk-downstream-automation
+
+#### Overlay Branch Directory Structures
+
+Today, we have the following files in the `ocp-release-operator-sdk` repo. This
+usually doesn't impact the merge but it is a set of files that are NOT upstream.
+So we will put these in an overlay branch that can be used to lay down over the
+mirrored upstream branch to create a downstream release branch.
+
+The following is a list of directories and files that will exist in the overlay
+branch:
+
+* ci
+  * dockerfiles
+    * variety of Dockerfiles used to build images for running tests
+* prow.Makefile # Makefile used by CI
+* patches
+  * contain the patch files that may be carried
+* release
+  * ansible
+    * ansible_collections # contains the sync ansible collections
+    * Dockerfiles # various dockerfiles
+  * helm
+    * Dockerfiles # various dockerfiles
+  * sdk
+    * Dockerfiles # various dockerfiles
+  * scorecard-test
+    * Dockerfiles # various dockerfiles
+  * scorecard-test-kuttl
+    * Dockerfiles # various dockerfiles
+* vendor
+  * contains the dependencies for the project
```

Comment:
I know this is a late drive-by, but (with the exception of `vendor`) would it be possible for us to put these in an obvious openshift-related root directory to avoid naming conflicts with upstream directories? For example: `github.com/openshift/operator-sdk/openshift/...`